lib/rubygems/exceptions.rb

```diff
@@ -20,6 +20,53 @@ class Gem::Exception < RuntimeError
 
 class Gem::CommandLineError < Gem::Exception; end
 
+class Gem::UnknownCommandError < Gem::Exception
+
+  class SpellChecker
+
+    attr_reader :error
+
+    def initialize(error)
+      @error = error
+    end
+
+    def corrections
+      @corrections ||=
+        spell_checker.correct(error.command_name).map(&:inspect)
+    end
+
+    private
+
+    def spell_checker
+      DidYouMean::SpellChecker.new(dictionary: error.command_names)
+    end
+
+  end
+
+  attr_reader :command_names, :command_name
+
+  def initialize(command_names, command_name)
+    @command_names = command_names
+    @command_name = command_name
+
+    self.class.correctable
+    super("Unknown command #{command_name}")
+  end
+
+  def self.correctable
+    @correctable ||=
+      begin
+        require 'did_you_mean'
```

Comment:
Does loading did_you_mean significantly reduce startup time?  There is some extra maintenance burden in lazy loading, especially when undocumented.

Reply:
To be honest I'm not 100% sure why it doesn't work if you try to require it at startup, but it doesn't. Also there's really no need to load it at all unless someone mistypes something. Otherwise you're going to force people to load it for a very small use case.

Reply:
What you mean by `It doesn't work`?

Reply:
Not sure anymore, this now does it immediately.